File: course-service/src/main/java/com/iarasantos/course_service/service/CourseService.java
package com.iarasantos.course_service.service;

import com.iarasantos.course_service.model.Course;

import java.util.List;

public interface CourseService {
    Course saveCourse(Course course);

    void deleteCourse(Long courseId);

    Course findCourseById(Long courseId);

    List<Course> findAllCourses();
}

File: frontend/src/services/course.service.js
import axios from "axios";
import { BASE_URL } from "../commom/Constants";
import { authHeader } from "./base.service";

const API_URL = BASE_URL + '/gateway/course'

class CourseService {

    saveCourse(course) {
        return axios.post(API_URL, course, {headers: authHeader()});
    }

    deleteCourse(course) {
        console.log({course})
        return axios.delete(API_URL + "/" + course.id, {headers: authHeader()})
    }

    getAllCourses() {
        return axios.get(API_URL);
    }
}

export default new CourseService();